nncf/quantization/algorithms/smooth_quant/torch_backend.py

```diff
@@ -0,0 +1,141 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Tuple
+
+import numpy as np
+
+import nncf.torch.graph.operator_metatypes as om
+from nncf.common.graph import NNCFGraph
+from nncf.common.graph import NNCFNode
+from nncf.common.graph.operator_metatypes import OperatorMetatype
+from nncf.common.graph.transformations.commands import TargetType
+from nncf.common.quantization.quantizer_propagation.structs import QuantizationTrait
+from nncf.common.tensor_statistics.statistic_point import StatisticPoint
+from nncf.experimental.common.tensor_statistics.collectors import MaxAggregator
+from nncf.experimental.common.tensor_statistics.collectors import TensorCollector
+from nncf.experimental.tensor import Tensor
+from nncf.openvino.graph.node_utils import get_channel_agnostic_reduction_axes
+from nncf.openvino.graph.transformations.commands import OVMultiplyInsertionCommand
+from nncf.openvino.graph.transformations.commands import OVWeightUpdateCommand
+from nncf.quantization.algorithms.smooth_quant.backend import SmoothQuantAlgoBackend
+from nncf.torch.graph.transformations.command_creation import create_command_to_update_weight
+from nncf.torch.graph.transformations.command_creation import multiply_insertion_command
+from nncf.torch.graph.transformations.commands import PTTargetPoint
+from nncf.torch.nncf_network import NNCFNetwork
+from nncf.torch.quantization.default_quantization import DEFAULT_PT_QUANT_TRAIT_TO_OP_DICT
+from nncf.torch.tensor_statistics.collectors import PTAbsMaxReducer
+from nncf.torch.tensor_statistics.collectors import PTNNCFCollectorTensorProcessor
+
+
+class PTSmoothQuantAlgoBackend(SmoothQuantAlgoBackend):
+    @property
+    def convolution_metatypes(self) -> List[OperatorMetatype]:
+        return [
+            om.PTConv1dMetatype,
+            om.PTConv2dMetatype,
+            om.PTConv3dMetatype,
+            om.PTModuleConv1dMetatype,
+            om.PTModuleConv2dMetatype,
+            om.PTModuleConv3dMetatype,
+            om.PTDepthwiseConv1dSubtype,
+            om.PTDepthwiseConv2dSubtype,
+            om.PTDepthwiseConv3dSubtype,
+            om.PTConvTranspose1dMetatype,
+            om.PTConvTranspose2dMetatype,
+            om.PTConvTranspose3dMetatype,
+        ]
+
+    @property
+    def matmul_metatypes(self) -> List[OperatorMetatype]:
+        return [om.PTMatMulMetatype, om.PTLinearMetatype, om.PTModuleLinearMetatype]
+
+    @property
+    def quantize_agnostic_metatypes(self) -> List[OperatorMetatype]:
+        return DEFAULT_PT_QUANT_TRAIT_TO_OP_DICT[QuantizationTrait.QUANTIZATION_AGNOSTIC]
+
+    @staticmethod
+    def target_point(target_node_name: str, port_id: int) -> PTTargetPoint:
+        return PTTargetPoint(TargetType.OPERATOR_PRE_HOOK, target_node_name, input_port_id=port_id)
+
+    @staticmethod
+    def is_node_with_weights(node: NNCFNode) -> bool:
+        return node.layer_attributes is not None
```

Comment:
I always return True right now because metatypes of matmuls and convolutions guarantee all nodes with the metatypes have weights, we can skip this check by returning True.